haystack/nodes/prompt/prompt_node.py

```diff
@@ -296,6 +296,7 @@ def invoke(self, *args, **kwargs):
         It takes a prompt and returns a list of generated text using the local Hugging Face transformers model
         :return: A list of generated text.
         """
+        self.pipe.model.eval()
```

Comment:
@sjrl this is not needed as pipeline automatically switches the model to eval

Reply:
I tried out the following code
```python
torch.manual_seed(1)
prompt_node = PromptNode("google/flan-t5-small", devices=["cpu"])
r = prompt_node("What is the capital of Germany?")
assert r[0].casefold() == "berlin"  # True

# Set underlying model to train mode
prompt_node.prompt_model.model_invocation_layer.pipe.model.train()
r2 = prompt_node("What is the capital of Germany?")
assert r2[0].casefold() == "berlin"  # False; Answer is "sterreich"
```
which shows that transformers also doesn't seem to set the model to eval mode at the call time of pipeline. 

Looking into their source code it seems they only set the model to eval mode at load time.

Reply:
Yes, I saw that too. But who is going to switch it to train? This way, we'll invoke eval every time we invoke the model, and that invocation needs to go through all network layers, which is in the case of LLMs a lot! It's not like that's going to kill the performance now, but I think we are being overcautious here.

Reply:
I've never seen a code example where someone makes sure that the model in the pipeline is switched to eval. Ok, sure, it makes sense that you switched it to eval in other cases - but not here - I think